### PR TITLE
Fix Che project validation & code refactoring

### PR DESCRIPTION
**Pull request by mkuznyetsov**

### CODE REVIEW

**Comment by ashumilova** [comment]
+1

**Comment by vitaliy-guliy** [comment]
Ok

File: plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectPresenter.java [hunk 1]
```diff
@@ -96,24 +95,34 @@ public void apply(List<ProjectDescriptor> projects) throws OperationException {
...
             view.showOsProjectNameError(locale.existingProjectNameError());
             return false;
         }
-        if (name.length() > 63
-            || name.isEmpty()
-            || !name.matches("[a-z0-9]([-a-z0-9]*[a-z0-9])?")) {
+        if (projectName.length() > 63
+            || projectName.length() < 2
+            || projectName.isEmpty()
+            || !projectName.matches("[a-z]([-a-z0-9]*[a-z0-9])?")) {
```
Comment (sleshchenko) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectPresenter.java]: "[a-z]([-a-z0-9]*[a-z0-9]" this line is used in two places. It would be better to make it as constant

File: plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java [hunk 2]
```diff
@@ -220,37 +215,31 @@ public void resetControls() {
...
         return osProjectDescriptionInput.getValue();
     }
 
-    /** {@inheritDoc} */
     @Override
     public String getCheProjectDescription() {
         return cheProjectDescriptionInput.getValue();
     }
 
-    /** {@inheritDoc} */
```
Comment (svor) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: Why did you delete blocks like this?

Comment (sleshchenko) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: Because it is redundant comments. Docs will be retrieved from super class's method automatically when @Override is present

Comment (azatsarynnyy) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: +1

File: plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectPresenter.java [hunk 3]
```diff
@@ -96,24 +95,34 @@ public void apply(List<ProjectDescriptor> projects) throws OperationException {
...
             return view.getExistedSelectedProject() != null && isCheProjectNameValid(view.getCheNewProjectName());
         }
     }
 
-    private boolean isOpenShiftProjectNameValid(String name) {
-        if (openShiftProjects.contains(name)) {
+    /**
+     * Checks whether project name matches OpenShift's naming requirements.
+     * If is not valid, display error message on the view, depending on case.
+     * @param projectName - project name
```
Comment (evoevodin) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectPresenter.java]: This javadoc is not formatted.
Params should be separated from javadoc description with one empty line, also `@param` description should be written in a new line without dash `-`.

See [formatter configuration](https://github.com/codenvy/codenvy-dev-resources/blob/master/che-codestyle/src/main/resources/che-codestyle-intellij-new.xml).
See javadoc [example](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#javadoc).

File: plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java (hunk 2 above)
Comment (svor) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: I think if we'll used some code analyzer (for example Sonar) for analyzing java doc coverage, they will say that this code doesn't have javadock. Or I'm wrong?

Comment (sleshchenko) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: I do not care

Comment (azatsarynnyy) [inline plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/project/wizard/page/configure/ConfigureProjectViewImpl.java]: I think it's not a good idea: slap @inheritDoc for Sonar only.

**Comment by sleshchenko** [comment]
OK
